client_app/src/components/shelter/RepeatableShifts.js

```diff
@@ -0,0 +1,207 @@
+// client_app/src/components/shelter/RepeatableShifts.js
+
+import React, { useState } from "react";
+import { useParams } from "react-router-dom";
+import "../../styles/shelter/RepeatableShifts.css";
+import { scheduleAPI } from "../../api/schedule";
+
+
+const RepeatableShifts = () => {
+  const { shelterId } = useParams(); // grab the shelterId from URL
+
+  const [shifts, setShifts] = useState([]);
+  const [newShift, setNewShift] = useState({
+    shiftName: "",
+    startTime: "",
+    endTime: "",
+    requiredVolunteers: 1,
+    maxVolunteers: 5,
+  });
+  const [successMessage, setSuccessMessage] = useState("");
+
+  const timeStringToMillis = (timeStr) => {
+    const [hours, minutes] = timeStr.split(":").map(Number);
+    return (hours * 60 + minutes) * 60 * 1000;
+  };
+
+  const handleInputChange = (e) => {
+    const { name, value } = e.target;
+    setNewShift({ ...newShift, [name]: value });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (newShift.startTime && newShift.endTime) {
+      const newEntry = {
+        ...newShift,
+        shift_start: timeStringToMillis(newShift.startTime),
+        shift_end: timeStringToMillis(newShift.endTime),
+      };
+      setShifts([...shifts, newEntry]);
+      setNewShift({
+        shiftName: "",
+        startTime: "",
+        endTime: "",
+        requiredVolunteers: 1,
+        maxVolunteers: 5,
```

Comment:
You have several places where code is duplicated (and since your field names are not consistent with what's expected, you'll end up having to change them in several places). Generalize your code so that there is no duplication.